### DISCUSSION TITLE
Allow accessing default values in flows filter trigger

### DISCUSSION THREAD

**Discussion by omkardedge011** [directus/directus#15246] [Feature Requests]
### Describe the Bug

Bug: Default value of field is not passed into the flow.

Following is the **students** collection data model.

- **roll_no** | Input - integer | Primary Key
- **name** | Input - string
- **class** | Input - string | default = A

![image](https://user-images.githubusercontent.com/75299473/186376331-2bd9a5e7-661e-4570-a583-aacb81c6edf3.png)

Following is the flow setup.

Description: After any item is created in **students** collection the data is logged to console. Trigger type is **action**(Non-Blocking).

![image](https://user-images.githubusercontent.com/75299473/186376752-b148982a-a7e5-4a2f-8b6a-2a0eeaa17ef7.png)

There are two cases:
1. Default value is changed while creating the item.

Here the **class** field is changed from default(**A**) to **B**.

![image](https://user-images.githubusercontent.com/75299473/186377531-c3f087b5-f46c-4dd7-a150-ff17f33b484c.png)

Payload after changing the default value and creating the item.
```
{
  "message": {
    "event": "students.items.create",
    "payload": {
      "name": "Mike",
      "class": "B"
    },
    "key": 41,
    "collection": "students"
  }
}
```

2. (**Bug**) Default value is not changed while creating the item.

Here the **class** field is default(**A**).

![image](https://user-images.githubusercontent.com/75299473/186379478-6b76c0f6-b154-4973-bdd9-24a837c2aba8.png)

Payload if default value is not changed and the item is created.
```
{
  "message": {
    "event": "students.items.create",
    "payload": {
      "name": "John"
    },
    "key": 42,
    "collection": "students"
  }
}
```

Further,
If the field is referenced inside the flow it shows **undefined**.

### To Reproduce

Following steps to reproduce:

(Have used **students** collection, you may use any).

1. Create a collection **students**.

- **roll_no** | Input - integer | Primary Key
- **name** | Input - string
- **class** | Input - string | default = A

![image](https://user-images.githubusercontent.com/75299473/186376331-2bd9a5e7-661e-4570-a583-aacb81c6edf3.png)

2. Create a flow with the same configuration as below.

![image](https://user-images.githubusercontent.com/75299473/186376752-b148982a-a7e5-4a2f-8b6a-2a0eeaa17ef7.png)

3. Create an item inside **students** collection without changing the default value.

4. Check the flow logs for the output.

### Errors Shown

Default value is not changed while creating the item.

Here the **class** field is default(**A**).

![image](https://user-images.githubusercontent.com/75299473/186379478-6b76c0f6-b154-4973-bdd9-24a837c2aba8.png)

Payload if default value is not changed and the item is created.
```
{
  "message": {
    "event": "students.items.create",
    "payload": {
      "name": "John"
    },
    "key": 42,
    "collection": "students"
  }
}
```

### What version of Directus are you using?

9.16.1

### What version of Node.js are you using?

16.13.2

### What database are you using?

MariaDB - 10.4.22

### What browser are you using?

Chrome

### How are you deploying Directus?

Running Locally

**Comment by Nitwel** [comment]
There are 3 possible solutions to this problem.

1. Provide default values exclusively in flows as they are required nowhere else
2. Expose the `directus_fields` collection in order to do some custom panel logic
3. Implement a `withDefault` operation, that adds all default values to an item

@rijkvanzanten thoughs?

**Comment by rijkvanzanten** [comment]
I'm not sure if this is a bug or a feature request 🤔 It is technically working as intended/expected, as the payload trigger is fired before default values are created. Besides, the default values are only added in the database, not beforehand, so with that in mind, you would never be able to get the default values _before_ the item is saved to the database. 

I'm wondering if we should instead allow a way for filter events / flows to easily retrieve the configured default values from the system tables, or if there's some other approach. I'll move this to Discussions so we can keep discussing 🙂

**Comment by battosai30** [comment]
In my mind, it's up the the user to decide what he/she wants to do with it. More informations is better than a one missing. If you don't nedd it, just don't use it.

"payload trigger is fired before default values are created" => it could be great to have an option to decide if we want or not to wait that "everything" is over before trigger ?

**Reply by rijkvanzanten** [reply]
> "payload trigger is fired before default values are created" => it could be great to have an option to decide if we want or not to wait that "everything" is over before trigger ?

The default value is a database feature. If you want to wait until "everything" is over, the value is already saved so you can't prevent the saving anymore in a filter hook 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
